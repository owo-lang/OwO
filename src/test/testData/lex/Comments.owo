{- {- Comments can be nested -} -}

{-
  {-
    Nested comments can cross lines
  -}
-}

{-
  {-
    Nested comments can cross lines
  --} This is a line comment
  (Line comments work inside block comments)
  -}
-}

{--
 -  This is a multi-line comment.
 -}

-- This is a single-line comment.

-- | An empty module.
module Hello where
