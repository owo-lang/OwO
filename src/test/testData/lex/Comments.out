CommentToken "{-" (0 1 1) (2 1 3)
CommentToken "{-" (3 1 4) (5 1 6)
CommentToken " Comments can be nested " (5 1 6) (29 1 30)
CommentToken "-}" (29 1 30) (31 1 32)
CommentToken "-}" (32 1 33) (34 1 35)
CommentToken "{-" (36 3 1) (38 3 3)
CommentToken "\n  " (38 3 3) (41 3 6)
CommentToken "{-" (41 4 3) (43 4 5)
CommentToken "\n    Nested comments can cross lines\n  " (43 4 5) (82 4 44)
CommentToken "-}" (82 6 3) (84 6 5)
CommentToken "-}" (85 7 1) (87 7 3)
CommentToken "{-" (89 9 1) (91 9 3)
CommentToken "\n  " (91 9 3) (94 9 6)
CommentToken "{-" (94 10 3) (96 10 5)
CommentToken "\n    Nested comments can cross lines\n  " (96 10 5) (135 10 44)
CommentToken "--} This is a line comment" (135 12 3) (161 12 29)
CommentToken "\n  (Line comments work inside block comments)\n  " (161 12 29) (209 12 77)
CommentToken "-}" (209 14 3) (211 14 5)
CommentToken "-}" (212 15 1) (214 15 3)
CommentToken "{-" (216 17 1) (218 17 3)
CommentToken "-" (218 17 3) (219 17 4)
CommentToken "\n " (219 17 4) (221 17 6)
CommentToken "-" (221 18 2) (222 18 3)
CommentToken "  This is a multi" (222 18 3) (239 18 20)
CommentToken "-" (239 18 20) (240 18 21)
CommentToken "line comment.\n " (240 18 21) (255 18 36)
CommentToken "-}" (255 19 2) (257 19 4)
CommentToken "-- This is a single-line comment." (259 21 1) (292 21 34)
CommentToken "-- | An empty module." (294 23 1) (315 23 22)
ModuleToken (316 24 1) (322 24 7)
IdentifierToken "Hello" (323 24 8) (328 24 13)
WhereToken (329 24 14) (334 24 19)
BraceLToken (335 25 1) (335 25 1)
BraceRToken (335 25 1) (335 25 1)
